app/models/mongoidable/abilities.rb

```diff
@@ -6,19 +6,36 @@ class Abilities
     include ::CanCan::Ability
     include Mongoidable::CaslList
 
-    attr_reader :ability_source
+    attr_reader :ability_source, :parent_model
```

Comment:
Add passing in of the parent_model
Add cache check and usage to can? and cannot?